qcodes/dataset/sqlite_base.py

```diff
@@ -209,10 +232,86 @@ def connect(name: str, debug: bool = False) -> sqlite3.Connection:
 
     if debug:
         conn.set_trace_callback(print)
+
+    init_db(conn)
+    perform_db_upgrade(conn, version=version)
```

Comment:
aha, so as i see, we don't modify the `_runs_table_schema` schema to have `guid` column so that `perform_db_upgrade` is called even for a fresh new database file?

Reply:
Yes, that is considered to be the best practice; the creation is always of version 0 and then upgrades are applied. That makes testing and partial upgrades (if you encounter some old version in the wild) very easy.

Reply:
i see. makes sense.